### PR TITLE
fix null payload bug

### PR DESCRIPTION
**Pull request by estezz**
I and a friend at another company had this issue.  https://github.com/pinterest/secor/issues/92

We are both running this fix.

### CODE REVIEW

File: src/main/java/com/pinterest/secor/message/Message.java [hunk 1]
```diff
@@ -20,55 +20,75 @@
...
 import java.lang.String;
 
+
 /**
  * Message represents a raw Kafka log message.
  *
  * @author Pawel Garbacki (pawel@pinterest.com)
  */
 public class Message {
+
```
Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/message/Message.java]: Remove extra blank lines.

Comment (estezz) [inline src/main/java/com/pinterest/secor/message/Message.java]: Sorry our cooperate standard has some double blank lines.  What is the best way for me to re-submit this.  I am pretty new to git-hub.

Leo

Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/message/Message.java]: You can make local changes and push to the review branch which I believe is estezz:master

**Comment by estezz** [comment]
Anyone know what that error is?

**Comment by pgarbacki** [comment]
@estezz any updates on fixing the build failure?

**Comment by estezz** [comment]
@pgarbacki I do not know how to replicate that error locally.  My fix should not have caused that error.  Can someone help me reproduce this locally?

**Comment by estezz** [comment]
@pgarbacki 
This passes integration the first check in.  Then I remove empty lines and now it is failing.  Might be environmental, how can we run it again?

**Comment by pgarbacki** [comment]
@estezz the sequence of commands is printed out in the log of the failed test: https://travis-ci.org/pinterest/secor/jobs/72136913

**Comment by llamahunter** [comment]
So, this pull seems 'bad', but the issue is real. Any update?

**Comment by estezz** [comment]
I've been busy at work, but got some time today to get some the integration
tests (run_tests.sh) going.  I'll try to debug it.  I made very little
change to secor, not sure why that broke the integration.


Leo

On Wed, Aug 19, 2015 at 10:49 AM, Richard Lee <notifications@github.com>
wrote:

> So, this pull seems 'bad', but the issue is real. Any update?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pinterest/secor/pull/112#issuecomment-132692069>.
>



-- 
Leo Woessner

Sign up for Uber at this link and we both get free rides :)
*https://www.uber.com/invite/ern6a <https://www.uber.com/invite/ern6a>*

**Comment by estezz** [comment]
Finally ready to merge.  Looks like the failure was environmental.

**Comment by HenryCaiHaiying** [comment]
The diff introduced lots of unneeded whitespace and empty lines, can you fix them?

Would replacing the null payload with empty array payload works for ThriftMessageParser?

        long timestamp = mDeserializer.partialDeserializeI64(message.getPayload(),
                new ThriftTemplate(mConfig.getMessageTimestampName()));

**Comment by estezz** [comment]
@HenryCaiHaiying I don't see any extra lines on my side.  I fixed that a while ago.  

This is working at Pearson and  at another company.  This has been in production for months now at Pearson.  

Can this be merged?

Leo Woessner

File: src/main/java/com/pinterest/secor/message/Message.java [hunk 2]
```diff
@@ -26,49 +26,60 @@
...
     private int mKafkaPartition;
     private long mOffset;
     private byte[] mPayload;
 
     protected String fieldsToString() {
-        return "topic='" + mTopic + '\'' +
-               ", kafkaPartition=" + mKafkaPartition +
-               ", offset=" + mOffset +
-               ", payload=" + new String(mPayload);
+
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/message/Message.java]: empty line here.

File: src/main/java/com/pinterest/secor/message/Message.java [hunk 3]
```diff
@@ -26,49 +26,60 @@
...
+	return "topic='" + mTopic + '\'' + ", kafkaPartition="
+		+ mKafkaPartition + ", offset=" + mOffset + ", payload="
+		+ new String(mPayload);
 
     }
 
     @Override
     public String toString() {
-        return "Message{" + fieldsToString() + '}';
+
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/message/Message.java]: empty line

File: src/main/java/com/pinterest/secor/message/Message.java [hunk 4]
```diff
@@ -26,49 +26,60 @@
...
+	return "Message{" + fieldsToString() + '}';
     }
 
     public Message(String topic, int kafkaPartition, long offset, byte[] payload) {
-        mTopic = topic;
-        mKafkaPartition = kafkaPartition;
-        mOffset = offset;
-        mPayload = payload;
+
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/message/Message.java]: empty line

File: src/main/java/com/pinterest/secor/message/Message.java [hunk 5]
```diff
@@ -26,49 +26,60 @@
...
 
     public String getTopic() {
 
-        return mTopic;
+	return mTopic;
     }
 
     public int getKafkaPartition() {
-        return mKafkaPartition;
+
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/message/Message.java]: empty line

File: src/main/java/com/pinterest/secor/message/Message.java [hunk 6]
```diff
@@ -26,49 +26,60 @@
...
+	return mKafkaPartition;
     }
 
     public long getOffset() {
-        return mOffset;
+
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/message/Message.java]: empty line

File: src/main/java/com/pinterest/secor/message/Message.java [hunk 7]
```diff
@@ -26,49 +26,60 @@
...
 
     public byte[] getPayload() {
-        return mPayload;
+
+	return mPayload;
     }
 
     public void write(OutputStream output) throws IOException {
-        output.write(mPayload);
+
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/message/Message.java]: empty line

File: src/main/java/com/pinterest/secor/message/Message.java [hunk 8]
```diff
@@ -26,49 +26,60 @@
...
 
     public long getOffset() {
-        return mOffset;
+
+	return mOffset;
     }
 
     public byte[] getPayload() {
-        return mPayload;
+
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/message/Message.java]: empty line

**Comment by HenryCaiHaiying** [comment]
See my detailed comment about the location of those empty lines.

**Comment by estezz** [comment]
Is this going to get merged or is there something still wrong?

**Comment by DayoOliyide** [comment]
:+1: for this to be merged.

**Comment by HenryCaiHaiying** [comment]
@estezz , I took a look again at the diff, you are still using TAB in your code, can you replace it with whitespace?

Does the null payload situation happens a lot?  If not, please Log a warning when you have null payload comes in.

**Comment by estezz** [comment]
I will change tabs to spaces.  But where are your coding standards.  I did not know you were sensitive to tabs, that was not pointed out.

**Comment by estezz** [comment]
The null payload could happen often depending on the environment

**Comment by HenryCaiHaiying** [comment]
@estezz I merged the changes, thanks for the contribution.  I don't know whether there is a coding guideline for this project since I am new here too.  But through all the companies I've worked through and  numerous coding guideline I've deal with, 'tab' is always prohibited since each developer's editor/IDE can map 'tab'  differently.

**Comment by llamahunter** [comment]
tabs in code are the devil's mischief.

**Pull request merged**